generator/src/main/java/com/scottlogic/deg/generator/walker/ReductiveDecisionTreeWalker.java

```diff
@@ -72,36 +69,40 @@ public ReductiveDecisionTreeWalker(
             StreamSupport.stream(
                 Spliterators.spliteratorUnknownSize(valueIterator, Spliterator.ORDERED),
                 false),
-            fieldValue -> getRowSpecsForFixedField(constraintNode, reductiveState, fixFieldStrategy));
+            fieldValue -> walkForNextValue(constraintNode, reductiveState, fixFieldStrategy));
     }
 
-    private Stream<RowSpec> getRowSpecsForFixedField(
+    private Stream<RowSpec> walkForNextValue(
```

Comment:
We discussed yesterday about clarifying the names for these methods; hence this name change. I'm not sure these name changes clarify anything further from before

Namely:
- `getRowSpecsForFixedField` &rarr; `walkForNextValue`
- `process` &rarr; `walkForNextField`

Both contain the verb _walk_ but this isn't what the _solver_ is doing; its reducing then selecting, how about - instead these names (as appropriate)

- ~`getRowSpecsForFixedField`~/~`walkForNextValue`~ &rarr; `reduce` or `prune` (_reduce_ does fit with the name (reductive) although _prune_ - if kept private - fits with the new (pruner) types
   - i.e. reduce/prune the tree for what is no longer valid
- ~`process`~/~`walkForNextField`~ &rarr; `fixField`
   - i.e. find the next field to fix

Reply:
Updated